### PR TITLE
Use `LocalDefId` for import IDs in trait map

### PR DESCRIPTION
**Pull request by petrochenkov**
cc https://github.com/rust-lang/rust/pull/73291#discussion_r439734867

### CODE REVIEW

File: src/librustc_resolve/lib.rs [hunk 1]
```diff
@@ -879,7 +879,7 @@ pub struct Resolver<'a> {
     /// `CrateNum` resolutions of `extern crate` items.
     extern_crate_map: FxHashMap<LocalDefId, CrateNum>,
     export_map: ExportMap<LocalDefId>,
-    trait_map: TraitMap<NodeId>,
+    trait_map: NodeMap<Vec<TraitCandidate>>,
```
Comment (marmeladema) [inline src/librustc_resolve/lib.rs:882]: If you use a `NodeMap` here, it means the keys are still `NodeId` and that you still need to call `node_id_to_hir_id` later.

**Comment by marmeladema** [comment]
Overall, its a step in the right direction but it solves only half of the problem since the key type if still `NodeId`.

**Comment by petrochenkov** [comment]
See https://github.com/rust-lang/rust/pull/73291#discussion_r439871713, this PR indeed solves only one half of the problem.

**Comment by marmeladema** [comment]
Great, ok with me then.

**Review by davidtwco — APPROVED** [review]
LGTM, r=me when CI passes

**Pull request merged**